tests/unit/test_utils.py

```diff
@@ -1748,15 +1750,54 @@ def test_deepcopy_returns_same_object(self):
         self.assertIs(botocore.UNSIGNED, copy.deepcopy(botocore.UNSIGNED))
 
 
+class RawResponse(BytesIO):
```

Comment:
I think we have like 5 of these mock raw response classes. After we merge this let's look into consolidating these. I've added a TODO for this in my before-send PR.

Reply:
Yep. I was hoping your `before-send` PR would get merged before I sent this up. So I pretty much copied and pasted what you had. I plan on using it once your PR gets merged.